core/src/syn/parser/thing.rs

```diff
@@ -208,12 +205,12 @@ impl Parser<'_> {
 
 				let next = self.peek_whitespace();
 				match next.kind {
-					t!(".") | TokenKind::Exponent | TokenKind::NumberSuffix(_) => {
+					t!(".") => {
 						// TODO(delskayn) explain that record-id's cant have matissas,
 						// exponents or a number suffix
-						unexpected!(self, next, "an integer");
+						unexpected!(self, next, "an integer", => "Numeric Record-id keys can only be integers");
```

Comment:
<img width="470" alt="image" src="https://github.com/user-attachments/assets/4730c00e-e0ee-4a43-8c77-7c9b51dee435">

Reply:
I can't replicate this screen shot. The first three queries seem reasonable to me, the last returns false when I run it, not true.

Reply:
But the last one should be true though? 

<img width="697" alt="Terminal 2024-09-11 13 31 05" src="https://github.com/user-attachments/assets/51c9e17c-c5c3-4bcc-94ef-aa37db7288b2">

Reply:
> The first three queries seem reasonable to me

In the original line, you state that only int record keys are allowed. I though one of the reasons for that was the same number (540 in this case) can be expressed in many ways (540, 54e1, 540e0, etc) and so it's not good that this validation doesn't really work

Reply:
It is because of old behavior and how record id keys are handled. When a record id key contains a non-digit character it is converted to a string. The error in the snippet can only be triggered when flexible record id's is disabled which is currently not used.

Reply:
do you think this should be changed in this pr or separately? Happy to help if needed

Reply:
I meant those should be equal as numbers not as record-id's `user:54e1 != user:540e0` as `user:"54e1" != user:"540e1"`. So the current behavior is correct.